src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -203,58 +217,69 @@ public IObservable<BranchTrackingDetails> CalculateHistoryDivergence(ILocalRepos
             {
-                var repo = gitService.GetRepository(repository.LocalPath);
-                var remoteName = repo.Head.RemoteName;
-                if (remoteName != null)
+                // BranchTrackingDetails doesn't keep a reference to Repository
+                using (var repo = gitService.GetRepository(repository.LocalPath))
                 {
-                    var remote = await gitClient.GetHttpRemote(repo, remoteName);
-                    await gitClient.Fetch(repo, remote.Name);
-                }
+                    var remoteName = repo.Head.RemoteName;
+                    if (remoteName != null)
+                    {
+                        var remote = await gitClient.GetHttpRemote(repo, remoteName);
+                        await gitClient.Fetch(repo, remote.Name);
+                    }
 
-                return Observable.Return(repo.Head.TrackingDetails);
+                    return Observable.Return(repo.Head.TrackingDetails);
+                }
             });
         }
 
         public IObservable<TreeChanges> GetTreeChanges(ILocalRepositoryModel repository, IPullRequestModel pullRequest)
         {
             return Observable.Defer(async () =>
             {
-                var repo = gitService.GetRepository(repository.LocalPath);
-                var remote = await gitClient.GetHttpRemote(repo, "origin");
-                await gitClient.Fetch(repo, remote.Name);
-                var changes = await gitClient.Compare(repo, pullRequest.Base.Sha, pullRequest.Head.Sha, detectRenames: true);
-                return Observable.Return(changes);
+                // TreeChanges doesn't keep a reference to Repository
+                using (var repo = gitService.GetRepository(repository.LocalPath))
+                {
+                    var remote = await gitClient.GetHttpRemote(repo, "origin");
+                    await gitClient.Fetch(repo, remote.Name);
+                    var changes = await gitClient.Compare(repo, pullRequest.Base.Sha, pullRequest.Head.Sha, detectRenames: true);
+                    return Observable.Return(changes);
+                }
             });
         }
 
         public IObservable<IBranch> GetLocalBranches(ILocalRepositoryModel repository, IPullRequestModel pullRequest)
         {
             return Observable.Defer(() =>
             {
-                var repo = gitService.GetRepository(repository.LocalPath);
-                var result = GetLocalBranchesInternal(repository, repo, pullRequest).Select(x => new BranchModel(x, repository));
-                return result.ToObservable();
+                // BranchModel doesn't keep a reference to repo
+                using (var repo = gitService.GetRepository(repository.LocalPath))
+                {
+                    var result = GetLocalBranchesInternal(repository, repo, pullRequest).Select(x => new BranchModel(x, repository));
+                    return result.ToObservable();
```

Comment:
@grokys I have a bad feeling about this one. What do you think?

Reply:
Yup, as expected!
![image](https://user-images.githubusercontent.com/11719160/33214560-575c0298-d124-11e7-9bc9-5f75b9a4f6a2.png)

Reply:
Hmm yeah, should either call `ToList()` on result here.

For reference, the other option is to  destroy the `repo` with a [`Using`](http://www.introtorx.com/content/v1.0.10621.0/11_AdvancedErrorHandling.html#Using) operator, I don't think that'd be necessary here however.